catalog/transformer/catalog-transformer-nitf/src/main/java/ddf/catalog/transformer/nitf/NitfInputTransformer.java

```diff
@@ -0,0 +1,608 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NitfInputTransformer.class);
+
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormat.
+            forPattern("yyyyMMddHHmmss")
+            .withZone(DateTimeZone.UTC);
+
+    private static final GeometryFactory GEOMETRY_FACTORY = new GeometryFactory(new PrecisionModel(
+            PrecisionModel.FLOATING), 4326);
+
+    private static final String NITF_VERSION = "ddf/catalog/transformer/nitf";
+
+    private MetacardFactory metacardFactory;
+
+    /**
+     * Transforms NITF images into a {@link Metacard}
+     */
+    public Metacard transform(InputStream input) throws IOException, CatalogTransformerException {
+        return transform(input, null);
+    }
+
+    public Metacard transform(InputStream input, String id)
+            throws IOException, CatalogTransformerException {
+        if (input == null) {
+            throw new CatalogTransformerException("Cannot transform null input.");
+        }
+
+        final DynamicMetacard metacard = metacardFactory.newInstance(Nitf.NAME);
+        if (metacard == null) {
+            throw new CatalogTransformerException("Unable to create NITF metacard.");
+        }
+
+        List<Polygon> polygonList = new ArrayList<>();
+        // TODO: update to XStream or some xml streaming library to create this metadata
+        StringBuilder metadataXml = new StringBuilder();
+        metadataXml.append("<metadata>\n");
+
+        try {
+            AllDataExtractionParseStrategy parsingStrategy = new AllDataExtractionParseStrategy();
+
+            NitfParserInputFlow inputFlow = new NitfParserInputFlow();
+            inputFlow.inputStream(input)
+                    .imageData()
+                    .fileHeader((header) -> {
+                        setHeaderAttributes(metacard, header);
+                    })
+                    .forEachImage((imagesegmentHeader, imageSegmentData) -> {
+                        handleImageSegmentHeader(metacard, imagesegmentHeader, polygonList);
+                        setMetadataFromImageSegments(metadataXml, imagesegmentHeader);
+                    });
+
+            // TODO: implement and use flow API for Graphic, Symbol, Label, and Text Segments
+
+            for (NitfGraphicSegmentHeader graphic : parsingStrategy.getGraphicSegmentHeaders()) {
+                setMetadataForGraphicSegment(metadataXml, graphic);
+            }
+            for (SymbolSegmentHeader symbol : parsingStrategy.getSymbolSegmentHeaders()) {
+                setMetadataForSymbolHeader(metadataXml, symbol);
+            }
+            for (LabelSegmentHeader label : parsingStrategy.getLabelSegmentHeaders()) {
+                setMetadataForLabelHeader(metadataXml, label);
+            }
+            for (TextSegmentHeader text : parsingStrategy.getTextSegmentHeaders()) {
+                setMetadataForTextHeader(metadataXml, text);
+            }
+
```

Comment:
'parsingStrategy' will always be empty because it was never used for parsing.  The flow API creates its own parsing strategy, so the NITF file header and image information will work fine.  These other for loops will never contain data.  imaging-nitf 0.3 will contain a more complete flow API, but until then you should probably revert to using NitfFileParser.parse(new NitfReader(input), parsingStrategy).  See the gist here: https://gist.github.com/dcruver/4c5354b2f2b11f878a1d